### PR TITLE
NR-145610: use release toolkit

### PR DESCRIPTION
**Pull request by sigilioso**
- [x] The branch of the PR is prefixed with renovate/*
- [x] Tests pass in renovate branches
- [x] security workflow is included and will pass in renovate branches
- [x] the changelog is validated in push_pr workflow
- [x] The repository has the secrets available
- [x] Release schedule is documented and not colliding
- [x] rt-update-changelog is in Makefile
- [x] Tags with leading v are safe
- [x] Common renovate config is pointed
- [x] Potential troublesome dependencies are checked
- [x] pre-release and release workflows are updated to notify failures
- [x] CHANGELOG.partial.md is in .gitignore
- [x] release-toolkit action excluded/included fields are set
- [x] Docker image pings go patch version
- [ ] The dependency dashboard is sanitized --> It will be enabled when the PR is merged, and then we need to remove Worktato comment and set `dependencies` label.

### CODE REVIEW

File: .github/workflows/on_prerelease.yml [hunk 1]
```diff
@@ -195,4 +195,16 @@ jobs:
           integration: nri-${{ env.INTEGRATION }}
           packageLocation: repo
           stagingRepo: true
-          upgrade: false
\ No newline at end of file
+          upgrade: false
+
+  notify-failure:
```
Comment (paologallinaharbur) [inline .github/workflows/on_prerelease.yml:200]: you forgot to delete the notification above

Comment (sigilioso) [reply .github/workflows/on_prerelease.yml:200]: 🤦 Thank you!

File: build/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20-buster
+FROM golang:1.20.7-bookworm
```
Comment (paologallinaharbur) [inline build/Dockerfile:1]: there is one also in the integrations tests, but I am not sure that you left it out on pourpose

**Review by paologallinaharbur — APPROVED** [review]
Thanks!

File: build/Dockerfile (hunk 2 above)
Comment (sigilioso) [reply build/Dockerfile:1]: I simply forgot it, let's keep them consistent: [7758c7d](https://github.com/newrelic/nri-haproxy/pull/112/commits/7758c7d55c1d76b03dd0a28ad4a7935f159907ea)

**Pull request merged**